src/topology/vector_bundle.lean

```diff
@@ -387,3 +411,57 @@ lemma is_open_map_proj : is_open_map Z.proj :=
 topological_fiber_bundle_core.is_open_map_proj Z
 
 end topological_vector_bundle_core
+
+end
+
+variable [∀ x, topological_space (E x)]
+
+open topological_space
+
+/-- This structure permits to define a vector bundle when trivializations are given as local
+equivalences but there is not yet a topology on the total space. The total space is hence given a
+topology in such a way that there is a fiber bundle structure for which the local equivalences
+are also local homeomorphism and hence vector bundle trivializations. -/
+@[nolint has_inhabited_instance]
+structure topological_vector_prebundle :=
+(trivialization_at : B → topological_vector_prebundle.trivialization R F E)
+(mem_base_trivialization_at : ∀ x : B, x ∈ (trivialization_at x).base_set)
+(continuous_triv_change : ∀ x y : B, continuous_on ((trivialization_at x) ∘
+  (trivialization_at y).to_local_equiv.symm) ((trivialization_at y).target ∩
+  ((trivialization_at y).to_local_equiv.symm ⁻¹' (trivialization_at x).source)))
+(inducing : ∀ (b : B), @inducing _ _ _ (⨆ x : B,
+  coinduced (trivialization_at x).to_prebundle_trivialization.set_symm (subtype.topological_space))
+  (λ x : (E b), (id ⟨b, x⟩ : total_space E)))
+
+namespace topological_vector_prebundle
+
+variables {R E F}
+
+/-- Natural identification as a `topological_fiber_prebundle`. -/
+def to_topological_fiber_prebundle (a : topological_vector_prebundle R F E) :
+  topological_fiber_prebundle F (proj E) :=
+{ trivialization_at := λ x, a.trivialization_at x, ..a }
+
+variables (a : topological_vector_prebundle R F E) (x : B)
+
+/-- Topology on the total space that will make the prebundle into a bundle. -/
+def total_space_topology (a : topological_vector_prebundle R F E) :
+  topological_space (total_space E) :=
+a.to_topological_fiber_prebundle.total_space_topology
+
+/-- Promotion from a `topologial_vector_prebundle.trivialization` to a
+  `topologial_vector_bundle.trivialization`. -/
+def vector_bundle_trivialization_at (a : topological_vector_prebundle R F E) (x : B) :
+  @topological_vector_bundle.trivialization R _ F E _ _ _ _ _ _ _ a.total_space_topology  :=
+begin
+  letI := a.total_space_topology,
+  exact { linear := (a.trivialization_at x).linear,
+  ..a.to_topological_fiber_prebundle.bundle_trivialization_at x }
+end
```

Comment:
A bit weird I admit

Reply:
Better way to do this?

Reply:
looks ok to me.

Reply:
Ok I thought it was better to avoid tactic mode in this case but if you say so...